core/src/core/transaction.rs

```diff
@@ -1613,6 +1771,23 @@ mod test {
 			excess_sig: sig.clone(),
 		};
 
+		// Test explicit protocol version.
+		for version in vec![ProtocolVersion(1), ProtocolVersion(2)] {
+			let mut vec = vec![];
+			ser::serialize(&mut vec, version, &kernel).expect("serialized failed");
+			let kernel2: TxKernel = ser::deserialize(&mut &vec[..], version).unwrap();
+			assert_eq!(
+				kernel2.features,
+				KernelFeatures::HeightLocked {
```

Comment:
can't we write code to allow direct kernel comparison as in assert_eq!(kernel, kernel2); ?
Or failing that, can't we write assert_eq!(kernel.features, kernel2.features); ?

Reply:
Yes we can do the following - 

    assert_eq!(kernel.features, kernel2.features);